crates/holochain/src/core/workflow/call_zome_workflow.rs

```diff
@@ -136,9 +137,38 @@ where
     .await?;
     tracing::trace!("After zome call");
 
-    inline_validation(workspace, network, conductor_api, Some(zome), ribosome).await?;
-
-    Ok(result)
+    let validation_result = inline_validation(
+        workspace.clone(),
+        network,
+        conductor_api,
+        Some(zome),
+        ribosome,
+    )
+    .await;
+    match validation_result {
+        Err(WorkflowError::SourceChainError(SourceChainError::InvalidCommit(_))) => {
+            let scratch_elements = workspace.source_chain().scratch_elements()?;
+            let is_locked_for_countersigning_session = if scratch_elements.len() == 1 {
+                let lock = SourceChain::lock_for_entry(scratch_elements[0].entry().as_option())?;
+                !lock.is_empty()
+                    && workspace
+                        .source_chain()
+                        .is_chain_locked(Vec::with_capacity(0))
+                        .await?
+                    && !workspace.source_chain().is_chain_locked(lock).await?
+            } else {
+                false
+            };
+            if is_locked_for_countersigning_session {
+                if let Err(error) = workspace.source_chain().unlock_chain().await {
+                    tracing::error!(?error);
+                }
+            }
+            validation_result?;
+            Ok(result)
+        }
+        _ => Ok(result),
```

Comment:
`validation_result` is not checked if it's not an invalid commit so all other errors from `inline_validation` are ignored here. 
```suggestion
        _ => {
            validation_result?;
            Ok(result)
        },
```